docs/concepts/terminology.md

```diff
@@ -0,0 +1,216 @@
+Pascal case and fixed width fonts are appropriate. We also tend to capitalise
+the former use, representing the feature in general, but fixed width fonts are
+not appropriate in that context.
+
+> Folks accustomed to Terraform might think of a Composition as a Terraform
+> module; the HCL code that describes how to take input variables and use them
+> to create resources in some cloud API. Folks accustomed to Helm might think of
+> a Composition as a Helm chart’s templates; the moustache templated YAML files
+> that describe how to take Helm chart values and render Kubernetes resources.
+
+### Composite Resource
+
+A “Composite Resource” or “XR” is an API type defined using Crossplane. A
+composite resource’s API type is arbitrary - dictated by the concept the author
+wishes to expose as an API, for example an “AcmeCoDB”. A common convention is
+for types to start with the word “Composite” - e.g. “CompositeAcmeCoDB”.
+
+We talk about Crossplane being a tool teams can use to define their own
+opinionated platform APIs. Those APIs are made up of composite resources; when
+you are interacting with an API that your platform team has defined, you’re
+interacting with composite resources.
+
+A composite resource can be thought of as the interface to a Composition. It
+provides the inputs a Composition uses to compose resources into a higher level
+concept. In fact, the composite resource _is_ the high level concept.
+
+The term “Composite Resource” refers to a class of types, so avoid using Pascal
+case - “Composite Resource” not CompositeResource. Use pascal case when
+referring to a distinct type of composite resource - e.g. a CompositeAcmeCoDB.
+
+> Folks accustomed to Terraform might think of a composite resource as a
+> `tfvars` file that supplies values for the variables a Terraform module uses
+> to create resources in some cloud API. Folks accustomed to Helm might think of
+> a composite resource as the `values.yaml` file that supplies inputs to a Helm
+> chart’s templates.
+
+### Composite Resource Claim
+
+A “Composite Resource Claim”, “XRC”, or just “a claim” is also an API type
+defined using Crossplane. Each type of claim corresponds to a type of composite
+resource, and the pair have nearly identical schemas. Like composite resources,
+the type of a claim is arbitrary.
+
+We talk about Crossplane being a tool platform teams can use to offer
+opinionated platform APIs to the application teams they support. The platform
+team offers those APIs using claims. It helps to think of the claim as an
+application team’s interface to a composite resource. You could also think of
+claims as the public (app team) facing part of the opinionated platform API,
+while composite resources are the private (platform team) facing part.
+
+A common convention is for a claim to be of the same type as its corresponding
+composite resource, but without the "Composite" prefix. So an "AcmeCoDB" would
```

Comment:
Maybe better?
```suggestion
A common convention is for a claim to have the same type name as its corresponding
composite resource, but without the "Composite" prefix. So an "AcmeCoDB" would
```

Reply:
Hrm, I want to avoid "type name" here since strictly it's the "kind" which makes up part of a type. I think I'll leave this as is.